packages/block-editor/src/components/global-styles/README.md

```diff
@@ -66,12 +76,115 @@ import { useGlobalSetting } from '@wordpress/block-editor';
+	const [ pColor, setPColor ] = useGlobalSetting(
+		'color.palette.theme',
+		'core/paragraph',
+		'base'
+	);
 
-	return "Something";
+	return 'Something';
 }
 ```
+
+## UI Components
+
+The global styles folder also offers a set of reusable UI components. These components follow a strict set of guidelines:
+
+-   They are independent of any context or any store dependency. They only rely on the props passed to them.
+-   They receive a similar set of props:
+
+    -   `value`: The value is a style object that maps closely the format used in `theme.json` and is also very close to the format of the `style` attributes for blocks. There are some differences with the block attributes due to the iteration process and the fact that we need to maintain compatibility with the existing blocks even if they predate the introduction of Global Styles and these UI components. An example value for a style object is:
+
+```js
+{
+  color: {
+    text: 'var:preset|color|blue',
+    background: '#FF0000',
+  },
+  typography: {
+    fontFamily: 'var:preset|font-family|base',
+    fontSize: '10px',
+    lineHeight: 1.5,
+  },
+  spacing: {
+    padding: 'var:preset|spacing|medium',
+  },
+  elements: {
+	link: {
+	  color: {
+		text: 'var:preset|color|green',
+	  },
+	},
+  },
+}
+```
+
+-   `onChange`: A callback that receives the new value and is called when the user changes the value of the UI component.
+-   `inheritedValue`: The inherited value is a style object that represents the combined value of the style inherited from the parent context in addition to the style applied to the current context. The format is the same as the `value` prop.
+-   `settings`: The settings are the theme.json settings. They are used to provide the UI components with the necessary information to render the UI. An example value for the settings is:
+
+```js
+{
+	color: {
+		palette: {
+			custom: [
+				{
+					slug: 'black',
+					color: '#000000',
+				},
+				{
+					slug: 'white',
+					color: '#FFFFFF',
+				},
+				{
+					slug: 'blue',
+					color: '#0000FF',
+				},
+			]
+		},
+		gradients: {
+			custom: [
+				{
+					slug: 'gradient-1',
+					gradient: 'linear-gradient(135deg,rgba(6,147,227,1) 0%,rgb(155,81,224) 100%)',
+				},
+				{
+					slug: 'gradient-2',
+					gradient: 'linear-gradient(135deg,rgba(6,147,227,1) 0%,rgb(155,81,224) 100%)',
+				},
+			]
+		},
+	},
+	typography: {
+		fontSizes: [
+			{
+				slug: 'small',
+				size: '12px',
+			},
+			{
+				slug: 'medium',
+				size: '16px',
+			},
+			{
+				slug: 'large',
+				size: '24px',
+			},
+		],
+	}
+}
+```
+-   `defaultControls`: The default controls are the controls that are used by default to render the UI. They are used to provide the UI components with the necessary information to render the UI. An example value for the default controls for the `ColorPanel` component is:
+
+```js
+{
+	background: true,
+	text: true,
+	link: true,
+},
+```
```

Comment:
Thanks for starting the discussion about what the defaults for each of the panels should be over in https://github.com/WordPress/gutenberg/pull/49389#discussion_r1162741177.

Something I was wondering about from that comment — should the defaults across the components be defaulting to `false` to follow the behaviour in `block.json` where we explicitly flag which ones we reveal? I know that in the site editor they're all displayed by default, but at the component level, I wasn't sure which idea would be preferable. Some example ideas:

* The component as it lives in the block editor package specifies the defaults across the board (e.g. most are hidden by default, except for child layout controls). It is then up to other use cases (e.g. global styles) to specify if they'd like to reveal more controls by default. Or:
* At the component level, we default to displaying everything by default. It is then up to the consumer to be more restrictive if it so wishes (e.g. the `block.json` behaviour).

What do you think?

Reply:
My strong opinion is that the behavior should be the same.
My other opinion that is a bit less strong is that there's also a third option which is for us to be opinionated about what the defaults should be, that way consumers of the component (think of a component as its own unit, independent of the existing usage) itself shouldn't have to provide that prop or a big object to actually choose what to show, they should have sane defaults. And since we control what is shown or hidden on Global Styles (as opposed to the blocks), I don't see why global styles shouldn't be using these sane defaults. (at least at the root level, as there's an argument that block level should also follow the block.json config)